core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java

```diff
@@ -332,6 +333,26 @@ private static int readCode(boolean[] rawbits, int startIndex, int length) {
     return res;
   }
 
+  private static byte boolsToByte(final boolean[] array, final int start) {
+    byte b = 0;
+    for (int i = 0; i < 8; i++) {
+      if (array[start + i]) {
+        b |= 1 << (7 - i);
+      } else {
+        throw new IndexOutOfBoundsException();
```

Comment:
I feel @iSDP and I have hit a dead end here, as we both don't know enough Java to fix this or throw a decent error. Until somebody figures this out, Or I learn enough, there is nothing I can do. Sorry about the trouble and thanks for listening.